File: public/stickers/quitar_fondo.py
"""
quitar_fondo.py
───────────────
Elimina el fondo blanco de todos los PNG en la carpeta de stickers.

USO:
  1. Guarda este archivo en tu Escritorio
  2. Abre PowerShell y ejecuta:
       pip install pillow numpy
       python C:\Users\marce\Desktop\quitar_fondo.py
"""

from PIL import Image
import os
import numpy as np

STICKERS_DIR = r"C:\Users\marce\Desktop\wilma-store\wilma-store\public\stickers"
TOLERANCIA   = 30   # 0=solo blanco puro · 50=más agresivo

def quitar_fondo_blanco(img, tolerancia=30):
    img  = img.convert("RGBA")
    data = np.array(img, dtype=np.float32)
    r, g, b, a = data[:,:,0], data[:,:,1], data[:,:,2], data[:,:,3]
    es_blanco = (
        (r > 255 - tolerancia) &
        (g > 255 - tolerancia) &
        (b > 255 - tolerancia) &
        (abs(r - g) < tolerancia) &
        (abs(r - b) < tolerancia) &
        (abs(g - b) < tolerancia)
    )
    data[:,:,3] = np.where(es_blanco, 0, a)
    return Image.fromarray(data.astype(np.uint8), "RGBA")

def procesar_carpeta(carpeta, tolerancia=30):
    archivos = [f for f in os.listdir(carpeta) if f.lower().endswith(".png")]
    if not archivos:
        print("No se encontraron PNG en:", carpeta)
        return
    print(f"Procesando {len(archivos)} stickers...\n")
    for nombre in archivos:
        ruta = os.path.join(carpeta, nombre)
        try:
            img = Image.open(ruta)
            resultado = quitar_fondo_blanco(img, tolerancia)
            resultado.save(ruta)
            print(f"  ✓ {nombre}")
        except Exception as e:
            print(f"  ✗ {nombre} — {e}")
    print("\n✅ Listo! Ahora sube los cambios:")
    print("   git add public/stickers/ && git commit -m 'Fix transparent stickers' && git push")

if __name__ == "__main__":
    procesar_carpeta(STICKERS_DIR, TOLERANCIA)

File: stickers/quitar_fondo.mjs
// quitar_fondo.mjs
// Elimina el fondo blanco de todos los stickers PNG
// USO: node quitar_fondo.mjs

import { readdir } from 'fs/promises'
import { dirname, join } from 'path'
import { fileURLToPath } from 'url'

const STICKERS_DIR = dirname(fileURLToPath(import.meta.url))
const TOLERANCIA = 30

// Instalar sharp si no está: npm install sharp
import sharp from 'sharp'

async function quitarFondo(inputPath) {
  const img   = sharp(inputPath)
  const meta  = await img.metadata()
  const { data, info } = await img
    .ensureAlpha()
    .raw()
    .toBuffer({ resolveWithObject: true })

  const pixels = new Uint8Array(data)
  const total  = info.width * info.height

  for (let i = 0; i < total; i++) {
    const idx = i * 4
    const r = pixels[idx]
    const g = pixels[idx + 1]
    const b = pixels[idx + 2]

    const esBlanco =
      r > 255 - TOLERANCIA &&
      g > 255 - TOLERANCIA &&
      b > 255 - TOLERANCIA &&
      Math.abs(r - g) < TOLERANCIA &&
      Math.abs(r - b) < TOLERANCIA &&
      Math.abs(g - b) < TOLERANCIA

    if (esBlanco) pixels[idx + 3] = 0  // transparente
  }

  await sharp(pixels, {
    raw: { width: info.width, height: info.height, channels: 4 }
  })
    .png()
    .toFile(inputPath + '.tmp.png')

  // Reemplazar original
  const { rename } = await import('fs/promises')
  await rename(inputPath + '.tmp.png', inputPath)
}

async function main() {
  const archivos = (await readdir(STICKERS_DIR)).filter(f => f.endsWith('.png'))
  console.log(`\nProcesando ${archivos.length} stickers...\n`)

  for (const archivo of archivos) {
    const ruta = join(STICKERS_DIR, archivo)
    try {
      await quitarFondo(ruta)
      console.log(`  ✓ ${archivo}`)
    } catch (e) {
      console.log(`  ✗ ${archivo} — ${e.message}`)
    }
  }
  console.log('\n✅ Listo! Ahora ejecuta:')
  console.log('   git add public/stickers/ && git commit -m "Fix transparent stickers" && git push')
}

main()
